mountpoint-s3/tests/fuse_tests/lookup_test.rs

```diff
@@ -198,3 +199,49 @@ fn lookup_unicode_keys_s3() {
 fn lookup_unicode_keys_mock() {
     lookup_unicode_keys_test(fuse::mock_session::new, "lookup_unicode_keys_test");
 }
+
+fn lookup_with_negative_cache<F>(creator_fn: F)
+where
+    F: FnOnce(&str, TestSessionConfig) -> (TempDir, BackgroundSession, TestClientBox),
+{
+    const FILE_NAME: &str = "hello.txt";
+    let config = TestSessionConfig {
+        filesystem_config: S3FilesystemConfig {
+            cache_config: CacheConfig {
+                serve_lookup_from_cache: true,
+                dir_ttl: Duration::from_secs(600),
+                file_ttl: Duration::from_secs(600),
+                ..Default::default()
+            },
+            ..Default::default()
+        },
+        ..Default::default()
+    };
+    let (mount_point, _session, mut test_client) = creator_fn("lookup_with_negative_cache", config);
+
+    // Check negative caching
+    let file_path = mount_point.path().join(FILE_NAME);
+    metadata(&file_path).expect_err("should fail as no object exists");
+
+    test_client.put_object(FILE_NAME, b"hello").unwrap();
+
+    metadata(&file_path).expect_err("should fail as mountpoint should use negative cache");
```

Comment:
could the cache entity expire at this point? or it relies on the delete-on-insert implementation detail?

Reply:
No, it relies on the 10 min TTL (line 213).